client/components/data/query-viewers/index.jsx

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { requestViewers } from 'calypso/state/viewers/actions';
+import { connect } from 'react-redux';
+
+class QueryViewers extends React.Component {
```

Comment:
Note: I tried using a functional component here but that resulted in unnecessary fetches (due to re-rendering issues I guess). After scrolling down the list it would fetch both next page and the page after that. Same for subsequent fetches. Using this class component we're comparing `page` (as trigger) explicitly which fixed this.

Reply:
No, I tried using a functional component first which didn't have a `getQuery` method. I have an idea in mind which I want to try in a follow up PR if that works for you?

Reply:
Works for me 👍